### PR TITLE
!entry cannot be null ever, so why even bother have a case for it?

### PR DESCRIPTION
**Pull request by ghost**
In the utilcode code, sometimes, there are checks for null right after a "new" statement, even though the new statement is not marked as no-throw.

As a result, it would be best to either remove the case, make it no throw, or do a try-catch.

I simply removed the case.

### CODE REVIEW

**Review by jkotas — APPROVED** [review]
Thanks

File: src/coreclr/src/md/ceefilegen/ceesectionstring.cpp [hunk 1]
```diff
@@ -67,8 +67,6 @@ CeeSectionString::~CeeSectionString()
 StringTableEntry* CeeSectionString::createEntry(__in_z LPWSTR target, ULONG hashId)
 {
     StringTableEntry *entry = new StringTableEntry;
```
Comment (jkotas) [inline src/coreclr/src/md/ceefilegen/ceesectionstring.cpp]: Actually, I think a better fix would be to use `new (nothrow)` here. The calling code does not seem to expect exceptions to be thrown from here.

**Review by jkotas — CHANGES_REQUESTED** [review]

**Comment by pi1024e** [comment]
Done @jkotas

**Review by jkotas — APPROVED** [review]
Thank you

**Comment by pi1024e** [comment]
Could we backport this to previous versions of Core being serviced now? @jkotas

**Comment by jkotas** [comment]
This fix does not meet the bar for back-porting to older versions.

**Pull request merged**